gravitee-apim-console-webui/src/shared/components/gio-role/gio-role.directive.spec.ts

```diff
@@ -0,0 +1,139 @@
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { TestBed, ComponentFixture } from '@angular/core/testing';
+import { Component, Input } from '@angular/core';
+
+import { GioRoleModule } from './gio-role.module';
+import { GioRoleCheckOptions } from './gio-role.directive';
+
+import { CurrentUserService } from '../../../ajs-upgraded-providers';
+import { User } from '../../../entities/user';
+
+@Component({ template: `<div *gioRole="role">A Content</div>` })
+class TestRoleComponent {
+  @Input()
+  role: GioRoleCheckOptions;
+}
+
+describe('GioPermissionDirective', () => {
+  let fixture: ComponentFixture<TestRoleComponent>;
+  const currentUser = new User();
+  currentUser.roles = [];
+
+  function prepareTestRoleComponent(role: GioRoleCheckOptions) {
+    fixture = TestBed.configureTestingModule({
+      declarations: [TestRoleComponent],
+      imports: [GioRoleModule],
+      providers: [{ provide: CurrentUserService, useValue: { currentUser } }],
+    }).createComponent(TestRoleComponent);
+
+    fixture.componentInstance.role = role;
+    fixture.detectChanges();
+  }
+
+  afterEach(() => {
+    currentUser.roles = [];
+  });
+
+  describe('anyOf', () => {
+    it('should hide element if role is not matching', () => {
+      currentUser.roles = [{ scope: 'ENVIRONMENT', name: 'ADMIN' }];
+
+      prepareTestRoleComponent({ anyOf: [{ scope: 'ORGANIZATION', name: 'ADMIN' }] });
+      fixture.detectChanges();
+
+      const inputEl = fixture.nativeElement.querySelector('div');
+      expect(inputEl).toBeNull();
+    });
+
+    it('should display element if role is matching', () => {
+      currentUser.roles = [
+        { scope: 'ENVIRONMENT', name: 'ADMIN' },
+        { scope: 'ORGANIZATION', name: 'ADMIN' },
+      ];
+
+      prepareTestRoleComponent({ anyOf: [{ scope: 'ORGANIZATION', name: 'ADMIN' }] });
+      fixture.detectChanges();
+
+      const inputEl = fixture.nativeElement.querySelector('div');
+      expect(inputEl).toBeDefined();
+    });
+
+    it('should display element if at least one role is matching', () => {
+      currentUser.roles = [
+        { scope: 'ENVIRONMENT', name: 'ADMIN' },
+        { scope: 'ORGANIZATION', name: 'ADMIN' },
+      ];
+
+      prepareTestRoleComponent({
+        anyOf: [
+          { scope: 'ORGANIZATION', name: 'ADMIN' },
+          { scope: 'ENVIRONMENT', name: 'USER' },
+        ],
+      });
+      fixture.detectChanges();
+
+      const inputEl = fixture.nativeElement.querySelector('div');
+      expect(inputEl).toBeDefined();
+    });
+  });
+
+  describe('noneOf', () => {
+    it('should display element if no permission is matching', () => {
```

Comment:
maybe rename the test to be clearer ?

Reply:
I reread it several times, and I couldn't think of a better name.

Reply:
it's ok then ;)